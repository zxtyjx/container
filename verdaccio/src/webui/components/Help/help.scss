.help {
  .noPkg {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 30px 0;
    font-size: 20px;
    color: #9f9f9f;

    .noPkgTitle {
      margin: 1em 0;
      padding: 0;
      font-size: 24px;
    }

    .noPkgIntro {
      line-height: 1.5;
      margin: 0 auto;
      font-size: 14px;
    }

    code {
      font-style: italic;
    }
  }
}
