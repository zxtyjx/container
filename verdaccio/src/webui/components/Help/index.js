import React from 'react';
import SyntaxHighlighter, {registerLanguage} from 'react-syntax-highlighter/dist/light';
import sunburst from 'react-syntax-highlighter/src/styles/sunburst';
import js from 'react-syntax-highlighter/dist/languages/javascript';

import classes from './help.scss';
import {getRegistryURL} from '../../utils/url';

registerLanguage('javascript', js);

const Help = () => {
  const registryURL = getRegistryURL();

    return (
      <div className={classes.help}>
        <li className={classes.noPkg}>
          <h1 className={classes.noPkgTitle}>
            No Package Published Yet
          </h1>
          <div className={classes.noPkgIntro}>
            <div>
              To publish your first package just:
            </div>
            <br/>
            <strong>
              1. Login
            </strong>
            <SyntaxHighlighter language='javascript' style={sunburst} id="adduser">
              {`npm adduser --registry  ${registryURL}`}
            </SyntaxHighlighter>
            <strong>2. Publish</strong>
            <SyntaxHighlighter language='javascript' style={sunburst} id="publish">
              {`npm publish --registry ${registryURL}`}
            </SyntaxHighlighter>
            <strong>3. Refresh this page!</strong>
          </div>
        </li>
      </div>
    );
};

export default Help;
