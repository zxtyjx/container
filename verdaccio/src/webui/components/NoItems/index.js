import React from 'react';
import PropTypes from 'prop-types';

import classes from './noItems.scss';

const NoItems = (props) => {
    return (
      <div className={classes.noItems}>
        <h2>{props.text}</h2>
      </div>
    );
};

NoItems.propTypes = {
  text: PropTypes.string.isRequired
};

export default NoItems;
