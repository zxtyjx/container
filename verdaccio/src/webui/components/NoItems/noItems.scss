.noItems {
  margin: 5em 0;
}
