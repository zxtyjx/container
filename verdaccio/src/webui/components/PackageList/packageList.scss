.pkgContainer {
  margin: 0;
  padding: 0;

  li {
    border-bottom: 1px solid #e4e8f1;
    list-style: none;

    &:last-child {
      border-bottom: none;
    }
  }

  .listTitle {
    font-weight: normal;
    font-size: 24px;
    margin-top: 30px;
    margin-bottom: 0;
  }
}
