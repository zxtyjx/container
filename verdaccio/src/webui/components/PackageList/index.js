import React from 'react';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';

import Package from '../Package';
import Help from '../Help';
import NoItems from '../NoItems';
import {formatAuthor, formatLicense} from '../../utils/package';

import classes from './packageList.scss';

export default class PackageList extends React.Component {
  static propTypes = {
    packages: PropTypes.array,
    help: PropTypes.bool
  };

  render() {
    return (
      <div className="package-list-items">
        <div className={classes.pkgContainer}>
          {this.renderTitle()}
          {this.isTherePackages() ? this.renderList() : this.renderOptions()}
        </div>
      </div>
    );
  }

  renderTitle() {
    if (this.isTherePackages() === false) {
      return;
    }

    return <h1 className={classes.listTitle}>Available Packages</h1>;
  }

  renderList() {
    return this.props.packages.map((pkg, i) => {
      const {name, version, description, time} = pkg;
      const author = formatAuthor(pkg.author);
      const license = formatLicense(pkg.license);
      return (
        <li key={i}>
          <Package {...{name, version, author, description, license, time}} />
        </li>
      );
    });
  }

  renderOptions() {
    if (this.isTherePackages() === false && this.props.help) {
      return this.renderHelp();
    } else {
      return this.renderNoItems();
    }
  }

  renderNoItems() {
    return (
      <NoItems
        className="package-no-items"
        text={'No items were found with that query'}
      />
    );
  }

  renderHelp() {
    if (this.props.help === false) {
      return;
    }
    return <Help />;
  }

  isTherePackages() {
    return isEmpty(this.props.packages) === false;
  }
}
