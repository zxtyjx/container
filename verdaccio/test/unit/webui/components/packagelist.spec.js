/**
 * PackageList component
 */

import React from 'react';
import { mount } from 'enzyme';
import PackageList from '../../../../src/webui/components/PackageList/index';
import Help from '../../../../src/webui/components/Help/index';
import NoItems from '../../../../src/webui/components/NoItems/index';
import { BrowserRouter } from 'react-router-dom';

describe('<PackageList /> component', () => {
  it('should load the component with no packages', () => {
    const props = {
      packages: [],
      help: true
    };
    const wrapper = mount(
      <PackageList packages={props.packages} help={props.help} />
    );

    const instance = wrapper.instance();
    expect(instance.isTherePackages()).toBeFalsy();
    expect(instance.renderHelp()).toBeTruthy();
    expect(instance.renderOptions()).toEqual(<Help />);
    expect(instance.renderTitle()).toBeUndefined();
  });

  it('should load the component with packages', () => {
    const props = {
      packages: [
        {
          name: 'verdaccio',
          version: '1.0.0',
          time: new Date(),
          description: 'Private NPM repository',
          author: { name: 'Sam' }
        },
        {
          name: 'abc',
          version: '1.0.1',
          time: new Date(),
          description: 'abc description',
          author: { name: 'Rose' }
        },
        {
          name: 'xyz',
          version: '1.1.0',
          description: 'xyz description',
          author: { name: 'Martin' }
        }
      ],
      help: false
    };
    const wrapper = mount(
      <BrowserRouter>
        <PackageList packages={props.packages} help={props.help} />
      </BrowserRouter>
    );

    const instance = wrapper.find(PackageList).instance();

    expect(instance.isTherePackages()).toBeTruthy();
    expect(instance.renderHelp()).toBeUndefined();
    expect(instance.renderTitle().props.children).toEqual('Available Packages');
    expect(instance.renderNoItems()).toEqual(
      <NoItems className="package-no-items" text="No items were found with that query" />
    );
    expect(instance.renderOptions()).toEqual(
      <NoItems className="package-no-items" text="No items were found with that query" />
    );
    // package count
    expect(wrapper.find('Package')).toHaveLength(3);
    // match snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });
});
