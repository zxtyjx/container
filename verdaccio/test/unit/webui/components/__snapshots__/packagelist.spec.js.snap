// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageList /> component should load the component with packages 1`] = `"<div class=\\"package-list-items\\"><div class=\\"pkgContainer\\"><h1 class=\\"listTitle\\">Available Packages</h1><li><section class=\\"package\\"><a href=\\"detail/verdaccio\\"><div class=\\"header\\"><div class=\\"title\\"><h1>verdaccio <span class=\\"el-tag el-tag--gray\\">v1.0.0</span></h1></div><div role=\\"author\\" class=\\"author\\">By: Sam</div></div><div class=\\"footer\\"><p class=\\"description\\">Private NPM repository</p></div><div class=\\"details\\"><div class=\\"homepage\\">Published less than a minute ago</div><div class=\\"license\\"></div></div></a></section></li><li><section class=\\"package\\"><a href=\\"detail/abc\\"><div class=\\"header\\"><div class=\\"title\\"><h1>abc <span class=\\"el-tag el-tag--gray\\">v1.0.1</span></h1></div><div role=\\"author\\" class=\\"author\\">By: Rose</div></div><div class=\\"footer\\"><p class=\\"description\\">abc description</p></div><div class=\\"details\\"><div class=\\"homepage\\">Published less than a minute ago</div><div class=\\"license\\"></div></div></a></section></li><li><section class=\\"package\\"><a href=\\"detail/xyz\\"><div class=\\"header\\"><div class=\\"title\\"><h1>xyz <span class=\\"el-tag el-tag--gray\\">v1.1.0</span></h1></div><div role=\\"author\\" class=\\"author\\">By: Martin</div></div><div class=\\"footer\\"><p class=\\"description\\">xyz description</p></div><div class=\\"details\\"><div class=\\"homepage\\"></div><div class=\\"license\\"></div></div></a></section></li></div></div>"`;
